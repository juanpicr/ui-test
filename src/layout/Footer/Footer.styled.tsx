import styled from 'styled-components'
import { Links } from 'components'

export const Footer = styled.footer`
  display: flex;
  align-items: flex-end;
  justify-content: space-between;
  margin: 0 1rem 3rem;
  @media (min-width: ${props => props.theme.md}) {
    align-items: flex-start;
  }
`

export const FooterLinks = styled(Links)`
  ul{
    @media (min-width: ${props => props.theme.md}) {
      display: flex;
    }
    li{
      padding: .5rem 0;
      font-size: 1.25rem;
      @media (min-width: ${props => props.theme.md}) {
        padding: 0;
        margin-right: 2rem;
        font-size: 1rem;
      }
      a{
        color: ${props => props.theme.darkGray};
        text-decoration: none;
        &:hover{
          color: ${props => props.theme.darkerGray};
          text-decoration: underline;
        }
      }
    }
  }
`

export const FooterSocial = styled(Links)`
  @media (min-width: ${props => props.theme.md}) {
    display: flex;
    align-items: center;
  }
  span{
    color: ${props => props.theme.darkerGray};
    font-size: 1.25rem;
    text-align: right;
    @media (min-width: ${props => props.theme.md}) {
      margin-right: 1rem;
      color: ${props => props.theme.darkGray};
      font-size: 0.833rem;
    }
  }
  & > ul{
    display: flex;
    margin-top: 1rem;
    @media (min-width: ${props => props.theme.md}) {
      margin-top: 3px;
    }
    & > li {
      margin-right: 1rem;
    }
  }
`
