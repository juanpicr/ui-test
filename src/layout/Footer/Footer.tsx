import React from 'react'

// Styled
import * as Styled from './Footer.styled'

// constants
import { footerLinks, footerSocial } from 'constants/index'

const Footer = () =>
  <Styled.Footer>
    <Styled.FooterLinks links={footerLinks}/>
    <Styled.FooterSocial links={footerSocial} title='Follow us' />
  </Styled.Footer>

export default Footer
