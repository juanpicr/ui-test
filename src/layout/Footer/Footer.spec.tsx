import * as React from 'react'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Footer from './Footer'

it('footer renders correctly', () => {
  const elem = renderer.create(<BrowserRouter><Footer/></BrowserRouter>).toJSON()
  expect(elem).toMatchSnapshot()
})
