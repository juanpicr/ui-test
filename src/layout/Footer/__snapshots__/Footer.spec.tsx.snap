// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer renders correctly 1`] = `
<footer
  className="sc-bczRLJ elECVD"
>
  <div
    className="sc-gsnTZi kATRWH"
  >
    <ul>
      <li>
        <a
          href="/termsAndConditions"
          onClick={[Function]}
        >
          Terms and Conditions
        </a>
      </li>
      <li>
        <a
          href="/privacyPolicy"
          onClick={[Function]}
        >
          Privacy Policy
        </a>
      </li>
      <li>
        <a
          href="/contactUs"
          onClick={[Function]}
        >
          Contact Us
        </a>
      </li>
    </ul>
  </div>
  <div
    className="sc-dkzDqf cuaJll"
  >
    <span>
      Follow us
    </span>
    <ul>
      <li>
        <a
          href="/https://www.facebook.com/"
          onClick={[Function]}
        >
          <img
            alt="facebook"
            src="facebook.svg"
          />
        </a>
      </li>
      <li>
        <a
          href="/https://www.twitter.com/"
          onClick={[Function]}
        >
          <img
            alt="twitter"
            src="twitter.svg"
          />
        </a>
      </li>
    </ul>
  </div>
</footer>
`;
